### PR TITLE
Update gRPC version to 1.60.2 to address vulnerability concerns and fix data plane integration tests

### PR DESCRIPTION
**Pull request by rohanshah18**
## Problem
1. The gRPC netty version 1.58.0 has vulnerability. 
2. I was still seeing some failing CI tests for data plane operations.

## Solution

Updated the following dependencies version from 1.58.0 to 1.60.2.
"io.grpc:grpc-protobuf"
"io.grpc:grpc-stub"
"io.grpc:grpc-netty"

Compatibility matrix for `grpc-netty` and `netty-tcnative-boringssl-static` can be found here: https://github.com/grpc/grpc-java/blob/master/SECURITY.md#netty

Also as a part of this PR, I have reduced the delay that is set after upserting vectors in the integration tests to 90 seconds instead of 120 seconds and for updateAndQueryPodTest, I have followed the same pattern as serverless indexes to upsert once and run all of the tests after the upsertion. 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [ ] Non-code change (docs, etc)
- [X] None of the above: (explain here)
Dependencies were updated so it doesn't qualify for bug fix but instead security fix.

## Test Plan

Ran integration tests on local a good couple of times and on the CI. Fixing configure index test is in flight and will be a future PR from @austin-denoble.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -46,16 +46,16 @@ dependencies {
     implementation 'org.slf4j:slf4j-api:2.0.5'
     implementation 'com.google.api.grpc:proto-google-common-protos:2.14.3'
     implementation 'com.squareup.okhttp3:okhttp:4.12.0'
+    implementation 'com.squareup.okhttp3:logging-interceptor:4.12.0'
```
Comment (rohanshah18) [inline build.gradle:49]: I should have updated this to 4.12.0 when updating `com.squareup.okhttp3:okhttp` to 4.12.0 in the previous PR

File: build.gradle [hunk 2]
```diff
@@ -46,16 +46,16 @@ dependencies {
...
     implementation 'com.fasterxml.jackson.core:jackson-core:2.14.2'
     implementation 'com.google.code.gson:gson:2.9.1'
-    implementation 'com.squareup.okhttp3:logging-interceptor:4.10.0'
     implementation 'io.gsonfire:gson-fire:1.8.5'
     implementation 'org.openapitools:jackson-databind-nullable:0.2.6'
     implementation 'com.google.protobuf:protobuf-java:3.25.2'
     compileOnly "org.apache.tomcat:annotations-api:6.0.53" // necessary for Java 9+
 
-    testImplementation("com.squareup.okhttp3:mockwebserver:4.10.0")
+    testImplementation("com.squareup.okhttp3:mockwebserver:4.12.0")
```
Comment (rohanshah18) [inline build.gradle:58]: I should have updated this to 4.12.0 when updating `com.squareup.okhttp3:okhttp` to 4.12.0 in the previous PR

File: src/integration/java/io/pinecone/helpers/IndexManager.java [hunk 3]
```diff
@@ -35,9 +35,10 @@ public static String findIndexWithDimensionAndType(Pinecone pinecone, int dimens
         }
 
         for (IndexModel indexModel : indexModels) {
-            if (indexModel.getDimension() == dimension
-                && (indexType.equalsIgnoreCase(IndexModelSpec.SERIALIZED_NAME_POD) && indexModel.getSpec().getPod() != null)
-                || (indexType.equalsIgnoreCase(IndexModelSpec.SERIALIZED_NAME_SERVERLESS) && indexModel.getSpec().getServerless() != null)
+            // Sparse-dense is only supported with DOTPRODUCT
```
Comment (rohanshah18) [inline src/integration/java/io/pinecone/helpers/IndexManager.java:38]: Even though we create new indexes of type DOT PRODUCT in this class when an index doesn't exist, the chances are there can be an index in console that was manually created or via other means but has a metric COSINE or EUCLIDEAN. This can be an issue for sparse-dense vectors tests since they are only supported with DOT PRODUCT metric

File: src/integration/java/io/pinecone/helpers/IndexManager.java [hunk 4]
```diff
@@ -63,7 +64,7 @@ public static String createNewIndex(Pinecone pinecone, int dimension, String ind
         CreateIndexRequest createIndexRequest = new CreateIndexRequest()
                 .name(indexName)
                 .dimension(dimension)
-                .metric(IndexMetric.DOTPRODUCT)
+                .metric(IndexMetric.DOTPRODUCT) // Sparse-dense is only supported with DOTPRODUCT
```
Comment (rohanshah18) [inline src/integration/java/io/pinecone/helpers/IndexManager.java:67]: Added comment for future reference on why we create index with DOTPRODUCT metric.

File: src/integration/java/io/pinecone/helpers/IndexManager.java (hunk 3 above)
Comment (austin-denoble) [reply src/integration/java/io/pinecone/helpers/IndexManager.java:38]: Is `findIndexWithDimensionAndType` only used for sparse-dense index tests? I'm not really understanding why we needed to add `indexModel.getMetric() == IndexMetric.DOTPRODUCT` to this.

File: src/integration/java/io/pinecone/integration/dataPlane/UpsertDescribeIndexStatsAndDeletePodTest.java [hunk 5]
```diff
@@ -23,56 +23,59 @@
...
         });
 
         // Delete 1 vector
         List<String> idsToDelete = new ArrayList<>(3);
-        idsToDelete.add(upsertIds.get(0));
-        indexClient.delete(idsToDelete, false, namespace, nullFilterStruct);
-        vectorCount -= idsToDelete.size();
-        int testSingleDeletedVectorCount = vectorCount;
+        String vectorIdToDelete = upsertIds.get(0);
+        idsToDelete.add(vectorIdToDelete);
```
Comment (austin-denoble) [inline src/integration/java/io/pinecone/integration/dataPlane/UpsertDescribeIndexStatsAndDeletePodTest.java:71]: Why the change pulling out `vectorIdToDelete`?

**Review by austin-denoble — APPROVED** [review]
Thanks for upgrading these dependency versions and reducing the wait time in these tests. 👍

File: src/integration/java/io/pinecone/integration/dataPlane/UpsertDescribeIndexStatsAndDeletePodTest.java (hunk 5 above)
Comment (rohanshah18) [reply src/integration/java/io/pinecone/integration/dataPlane/UpsertDescribeIndexStatsAndDeletePodTest.java:71]: We are just assigning upsertIds.get(0) to vectorIdToDelete which is still added to idsToDelete.

File: src/integration/java/io/pinecone/helpers/IndexManager.java (hunk 3 above)
Comment (rohanshah18) [reply src/integration/java/io/pinecone/helpers/IndexManager.java:38]: Is Cosine or Euclidean metric stricly needed for any other tests?
